gobblin-utility/src/main/java/gobblin/util/concurrent/TaskSchedulerFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.concurrent;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Enums;
+import com.google.common.base.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+public class TaskSchedulerFactory {
+  @AllArgsConstructor
+  private enum TaskSchedulerType {
+    SCHEDULEDEXECUTORSERVICE(ScheduledExecutorServiceTaskScheduler.class),
+    HASHEDWHEELTIMER(HashedWheelTimerTaskScheduler.class);
+
+    @Getter
+    private final Class<? extends TaskScheduler> taskSchedulerClass;
+
+    public static Optional<TaskSchedulerType> parse(Optional<String> name) {
```

Comment:
What is the point of returning an optional here? Couldn't you just return the default? (that seems to be the only way you are using `parse`)